grype/search/language.go

```diff
@@ -20,7 +20,12 @@ func ByPackageLanguage(store vulnerability.ProviderByLanguage, p pkg.Package, up
 		return nil, fmt.Errorf("matcher failed to fetch language=%q pkg=%q: %w", p.Language, p.Name, err)
 	}
 
-	applicableVulns, err := onlyVulnerableVersions(verObj, allPkgVulns)
+	applicableVulns, err := onlyQualifiedPackages(p, allPkgVulns)
+	if err != nil {
+		return nil, fmt.Errorf("unable to filter language-related vulnerabilities: %w", err)
+	}
+
+	applicableVulns, err = onlyVulnerableVersions(verObj, applicableVulns)
```

Comment:
nit: it would be nice to add a TODO here that states we want to port over this check to be a qualifier (and removed)